clients.go

```diff
@@ -330,7 +330,7 @@ func (cl *Client) ResendInflightMessages(force bool) error {
 func (cl *Client) ClearInflights(now, maximumExpiry int64) []uint16 {
 	deleted := []uint16{}
 	for _, tk := range cl.State.Inflight.GetAll(false) {
-		if (tk.Expiry > 0 && tk.Expiry < now) || tk.Created+maximumExpiry < now {
+		if (tk.Expiry > 0 && tk.Expiry < now) || tk.Created < now-maximumExpiry {
```

Comment:
This addresses the issue in #306 but I feel that this change may be harder to understand without context. Also, if maximum expiry is maxInt, we don't need to evaluate this on every iteration. 

I'll look at this whole issue more deeply tomorrow as I should have some time 👍🏻

Reply:
@mochi-co Using addition here might lead to integer overflow, while subtraction can prevent it.

Reply:
@werbenhu I agree with you, however I just want to make sure we have the most considered solution as it was not clear to me when I first read it

Reply:
You're right; perhaps we can use another, clearer approach to address the overflow issue.